libs/ledger-live-common/src/families/aptos/getFeesForTransaction.ts

```diff
@@ -12,63 +12,60 @@ type IGetEstimatedGasReturnType = {
   estimate: {
     maxGasAmount: string;
     gasUnitPrice: string;
-    sequenceNumber: string;
-    expirationTimestampSecs: string;
   };
   errors: TransactionErrors;
 };
 
-const CACHE = new Map();
+const CACHE = {
+  amount: new BigNumber(0),
+  estimate: Promise.resolve({
+    fees: new BigNumber(0),
+    estimate: {
+      maxGasAmount: "",
+      gasUnitPrice: "",
+    },
+    errors: {},
+  }),
+};
 
 export const getFee = async (
   account: Account,
   transaction: Transaction,
   aptosClient: AptosAPI,
 ): Promise<IGetEstimatedGasReturnType> => {
   const res = {
-    fees: new BigNumber(0),
+    fees: new BigNumber(DEFAULT_GAS).multipliedBy(DEFAULT_GAS_PRICE),
     estimate: {
-      maxGasAmount: transaction.estimate.maxGasAmount,
-      gasUnitPrice: transaction.estimate.gasUnitPrice,
-      sequenceNumber: transaction.options.sequenceNumber || "",
-      expirationTimestampSecs: transaction.options.expirationTimestampSecs || "",
+      maxGasAmount: DEFAULT_GAS.toString(),
+      gasUnitPrice: DEFAULT_GAS_PRICE.toString(),
     },
     errors: { ...transaction.errors },
   };
 
-  let gasPrice = DEFAULT_GAS_PRICE;
   let gasLimit = DEFAULT_GAS;
-  let sequenceNumber = "";
-
-  try {
-    const { gas_estimate } = await aptosClient.estimateGasPrice();
-    gasPrice = gas_estimate;
-  } catch (err) {
-    // skip
-  }
-
+  let gasPrice = DEFAULT_GAS_PRICE;
+  transaction.options = {
+    maxGasAmount: gasLimit.toString(),
+    gasUnitPrice: gasPrice.toString(),
+  };
   if (account.xpub) {
     try {
       const publicKeyEd = new Ed25519PublicKey(account.xpub as string);
       const tx = await buildTransaction(account, transaction, aptosClient);
       const simulation = await aptosClient.simulateTransaction(publicKeyEd, tx);
       const completedTx = simulation[0];
 
-      const expectedGas = BigNumber(gasLimit * gasPrice);
-      const isUnderMaxSpendable = !transaction.amount
+      gasLimit = new BigNumber(completedTx.gas_used).multipliedBy(ESTIMATE_GAS_MUL);
+      gasPrice = new BigNumber(completedTx.gas_unit_price);
+
+      const expectedGas = gasPrice.multipliedBy(gasLimit);
+
+      const isUnderMaxSpendable = transaction.amount
```

Comment:
We can merge a first iteration but I still don't understand why we need to check compared to account spendable balance here

This is the job of prepareTransaction and estimateMaxSpendableBalance